version: "3.8"

volumes:
  conda_store_data:

services:
  initializer:
    image: alpine
    restart: "no"
    # hack to set perimssions on volume
    entrypoint: |
      /bin/sh -c "chown -R 1000:1000 /var/lib/conda-store/"
    volumes:
      - conda_store_data:/var/lib/conda-store/

  conda-store-server:
    build: ../../conda-store-server
    user: 1000:1000
    depends_on:
      initializer:
        condition: service_completed_successfully
    volumes:
      - conda_store_data:/var/lib/conda-store/
      - ./conda_store_config.py:/opt/conda_store/conda_store_config.py:ro
    healthcheck:
      test: ["CMD", "curl", "--fail", "http://localhost:5000/conda-store/api/v1/"]
      interval: 10s
      timeout: 5s
      retries: 5
    platform: linux/amd64
    command: ['conda-store-server', '--config', '/opt/conda_store/conda_store_config.py', '--standalone=true']
    ports:
      - "5000:5000"
